src/probnum/diffeq/odefiltsmooth/odefiltsmooth.py

```diff
@@ -218,196 +254,187 @@ def probsolve_ivp(
+def _create_filter(ivp, prior, method):
     """Create the solver object that is used."""
-    if method not in ["ekf0", "ekf1", "ukf", "eks0", "eks1", "uks"]:
-        raise ValueError("Method not supported.")
-    gfilt = _string2filter(ivp, prior, method, **kwargs)
+    if method not in ["EK0", "EK1", "UK"]:
+        raise ValueError("This method is not supported.")
+    gfilt = _string2filter(ivp, prior, method)
     return gfilt
 
 
-def _create_steprule(atol, rtol, step, firststep, ivp):
-    _check_step_tol(step, atol, rtol)
+def _create_steprule(atol, rtol, step, firststep):
+    if atol is None and rtol is None and step is None:
+        errormsg = (
+            "Please specify either absolute and relative tolerances or a step size."
+        )
+        raise ValueError(errormsg)
 
-    if step is not None:
+    if atol is None and rtol is None:
         stprl = steprule.ConstantSteps(step)
     else:
-        if firststep is None:
-            # lazy version of Hairer, Wanner, Norsett, p. 169
-            norm_y0 = np.linalg.norm(ivp.initrv.mean)
-            norm_dy0 = np.linalg.norm(ivp(ivp.t0, ivp.initrv.mean))
-            firststep = 0.01 * norm_y0 / norm_dy0
+        firststep = step if step is not None else firststep
         stprl = steprule.AdaptiveSteps(firststep=firststep, atol=atol, rtol=rtol)
     return stprl
 
 
-def _check_step_tol(step, atol, rtol):
-    both_none = atol is None and rtol is None and step is None
-    both_not_none = (atol is not None and rtol is not None) and step is not None
-    if both_none or both_not_none:
-        errormsg = "Please specify either a tolerance or a step size."
-        raise ValueError(errormsg)
-    atol_not_rtol = atol is not None and rtol is None
-    rtol_not_atol = rtol is not None and atol is None
-    if atol_not_rtol or rtol_not_atol:
-        errormsg = "Please specify either both atol and rtol, or neither."
-        raise ValueError(errormsg)
-
-
-def _string2prior(ivp, which_prior, **kwargs):
+def _string2prior(ivp, which_prior, driftspeed, lengthscale):
 
-    ibm_family = ["ibm1", "ibm2", "ibm3", "ibm4"]
-    ioup_family = ["ioup1", "ioup2", "ioup3", "ioup4"]
-    matern_family = ["matern32", "matern52", "matern72", "matern92"]
+    ibm_family = ["IBM1", "IBM2", "IBM3", "IBM4"]
+    ioup_family = ["IOUP1", "IOUP2", "IOUP3", "IOUP4"]
+    matern_family = ["MAT32", "MAT52", "MAT72", "MAT92"]
     if which_prior in ibm_family:
-        return _string2ibm(ivp, which_prior, **kwargs)
+        return _string2ibm(ivp, which_prior)
     elif which_prior in ioup_family:
-        return _string2ioup(ivp, which_prior, **kwargs)
+        return _string2ioup(ivp, which_prior, driftspeed=driftspeed)
     elif which_prior in matern_family:
-        return _string2matern(ivp, which_prior, **kwargs)
-    else:
-        raise RuntimeError("It should have been impossible to reach this point.")
+        return _string2matern(ivp, which_prior, lengthscale=lengthscale)
```

Comment:
this could in principle be replaced with `if which_prior[:3] == "IBM": ...`

Reply:
the same obviously holds for the other priors (which is also true for the comment below)

Reply:
I'm not a fan of the fact that we match the order here, even though we just want to know the prior. On the other hand, matching the full string is cleaner than matching substrings, so idk....